File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    obj = {}
    
    for item in completion:
        if item in obj:
            obj[item] += 1
        else:
            obj[item] = 1
    
    # print(obj)
    for target in participant:
        if target not in obj:
            return target
        else: 
            if obj[target] >= 1:
                obj[target] -= 1
            elif obj[target] == 0:
                return target
    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    const completionRef = {};
    
    participant.forEach((person) => {
        completionRef[person] = completionRef[person] ? completionRef[person] + 1 : 1;
    });
    
    completion.forEach((person) => {
        completionRef[person] -= 1
    });
    
    for (const [key, value] of Object.entries(completionRef)) {
        if (value === 1) {
            answer = key;
            break;
        }
    }
    
    return answer;
}